src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/CodeGeneration/CodeWriterExtensions.cs

```diff
@@ -544,7 +540,14 @@ public static CSharpCodeWritingScope BuildClassDeclaration(
 
             if (hasInterfaces)
             {
-                writer.Write(string.Join(", ", interfaces));
+                for (var i = 0; i < interfaces.Count; i++)
+                {
+                    WriteOptionalToken(interfaces[i]);
+                    if (i != interfaces.Count - 1)
+                    {
+                        writer.Write(", ");
+                    }
+                }
```

Comment:
Small hit, think this would be a bit clearer as:
```suggestion
                WriteOptionalToken(interfaces[0]);
                for (var i = 1; i < interfaces.Count; i++)
                {
                    writer.Write(", ");
                    WriteOptionalToken(interfaces[i]);
                }
```

Reply:
Also, when are these going to be optional tokens?